src/main.ts

```diff
@@ -0,0 +1,174 @@
+import { Actor, log } from 'apify';
+import { BasicCrawler, BasicCrawlingContext } from 'crawlee';
+import snowflake from 'snowflake-sdk';
+import fs, { createWriteStream } from 'fs';
+import { CrawlerContext, Input } from './types.js';
+import streamJson from 'stream-json';
+import { promisifySnoflakeExecute } from './helpers.js';
+import stringerModule from 'stream-json/Stringer.js';
+import disassemblerModule from 'stream-json/Disassembler.js';
+import streamChain from 'stream-chain';
+import flat from 'flat';
+import streamArrayModule from 'stream-json/streamers/StreamArray.js';
+// commonJS modules
+const { flatten } = flat;
+const { streamArray } = streamArrayModule;
+const { chain } = streamChain;
+const { parser } = streamJson;
+const { stringer } = stringerModule;
+const { disassembler } = disassemblerModule;
+
+await Actor.init();
+
+const input = (await Actor.getInput<Input>())!;
+const {
+    synchronizeSchema,
+    transformJsonKeyFunction,
+    datasetId,
```

Comment:
Let's also add optional `resource` which is one the default fields you get from a webhook and you can check if there is `resource.defaultDatasetId` to work after default webhook